pkg/operator/ceph/cluster/nodedaemon/exporter.go

```diff
@@ -177,7 +177,7 @@ func getCephExporterDaemonContainer(cephCluster cephv1.CephCluster, cephVersion
 	container := corev1.Container{
 		Name:            "ceph-exporter",
 		Command:         []string{"ceph-exporter"},
-		Args:            []string{"--sock-dir", sockDir, "--port", strconv.Itoa(int(DefaultMetricsPort)), "--prio-limit", perfCountersPrioLimit, "--stats-period", statsPeriod},
+		Args:            []string{"--sock-dir", sockDir, "--addrs", "::", "--port", strconv.Itoa(int(DefaultMetricsPort)), "--prio-limit", perfCountersPrioLimit, "--stats-period", statsPeriod},
```

Comment:
I think we should check `dualStack` or `ipFamily` if `dualStack` is set to `false` instead of defaulting it to `IPv6` always.
```suggestion
		exporterAddrs := "0.0.0.0"
	if cephCluster.Spec.Network.DualStack || cephCluster.Spec.Network.IPFamily == "IPv6" {
		exporterAddrs = "::"
	}

	envVars := append(
		controller.DaemonEnvVars(cephCluster.Spec.CephVersion.Image),
		v1.EnvVar{Name: "CEPH_ARGS", Value: fmt.Sprintf("-m $(ROOK_CEPH_MON_HOST) -k %s", keyring.VolumeMount().AdminKeyringFilePath())})

	container := corev1.Container{
		Name:            "ceph-exporter",
		Command:         []string{"ceph-exporter"},
		Args:            []string{"--sock-dir", sockDir, "--addrs", exporterAddrs, "--port", strconv.Itoa(int(DefaultMetricsPort)), "--prio-limit", perfCountersPrioLimit, "--stats-period", statsPeriod},
		Image:           cephImage,
		ImagePullPolicy: controller.GetContainerImagePullPolicy(cephCluster.Spec.CephVersion.ImagePullPolicy),
		Env:             envVars,
		VolumeMounts:    volumeMounts,
		Resources:       cephv1.GetCephExporterResources(cephCluster.Spec.Resources),
		SecurityContext: controller.PodSecurityContext(),
	}
```

Reply:
I was thinking about doing something similar but didn't after I noticed it listened on both v4 and v6.  I'll apply the requested changes shortly 🙂

Reply:
> Thanks! Btw just a small update , I mistakenly mentioned of initializing `exporterAddrs := "0.0.0.0"` which isnt really required, bcz exporter daemon by default anyways uses this value.

I will omit the flag entirely unless IPv6 or Dual Stack is enabled then